### PR TITLE
Refactor PrescribedAtmosphere and add an idealized atmosphere simulation

### PR DESCRIPTION
**Pull request by glwagner**
This PR refactors `PrescribedAtmosphere` so that we can set up ocean simulations beneath idealized atmospheres.

I did the minimum possible in this PR, which basically preserves the kind of atmosphere that JRA55 uses --- everything as a FieldTimeSeries. Even to get this to work I needed to make some changes, including to `Oceananigans`. So it depends on https://github.com/CliMA/Oceananigans.jl/pull/3928

I also had to use a hacked version of `OrthogonalSphericalShellGrids` (#glw/fix) since it's broken for Oceananigans 0.94 currently.

I added an example called `idealized_single_column_simulation.jl` which evolves a linearly stratified ocean beneath constant 10 m/s winds at 20 deg C and with 400 W / m^2 solar insolation. It's interesting to see the difference between this and the typical idealized forcing that we use --- with constant winds, there is significant cooling and evaporation. Also, the fluxes decrease in time because 1) the ocean cools and 2) the ocean speeds up, decreasing the momentum flux in a relative winds bulk formulation, which in turn decreases the latent heat flux because the latent heat flux is proportional to the friction velocity.

Here's a movie of the results:

https://github.com/user-attachments/assets/1837bf42-9fda-42e7-b562-77ba0a7167cb

Here's the single column setup:

```julia
using ClimaOcean
using Oceananigans
using Oceananigans.Units
using SeawaterPolynomials

# Ocean state parameters
T₀ = 20   # Surface temperature, ᵒC
S₀ = 35   # Surface salinity
N² = 1e-4 # Buoyancy gradient due to temperature stratification
f = 0     # Coriolis parameter

# Atmospheric state parameters
Tₐ = 273.15 + 20 # Kelvin
u₁₀ = 10 # wind at 10 m, m/s
qₐ = 0.01 # specific humidity
Qℓ = 400 # shortwave radiation (W m⁻², positive means heating right now)

# Build the atmosphere
atmosphere_grid = RectilinearGrid(size=(), topology=(Flat, Flat, Flat))
atmosphere_times = range(0, 1day, length=3)
atmosphere = PrescribedAtmosphere(atmosphere_grid, atmosphere_times)

parent(atmosphere.tracers.T) .= Tₐ     # K
parent(atmosphere.velocities.u) .= u₁₀ # m/s
parent(atmosphere.tracers.q) .= qₐ     # mass ratio
parent(atmosphere.downwelling_radiation.shortwave) .= Qℓ # W

# Build ocean model at rest with initial temperature stratification
grid = RectilinearGrid(size=100, z=(-400, 0), topology=(Flat, Flat, Bounded))
ocean = ocean_simulation(grid, coriolis=FPlane(; f))

eos = ocean.model.buoyancy.model.equation_of_state
g = ocean.model.buoyancy.model.gravitational_acceleration
α = SeawaterPolynomials.thermal_expansion(T₀, S₀, 0, eos)
dTdz = N² / (α * g)
Tᵢ(z) = T₀ + dTdz * z
set!(ocean.model, T=Tᵢ, S=S₀)

radiation = Radiation(ocean_albedo=0.1)
model = OceanSeaIceModel(ocean; atmosphere, radiation)
simulation = Simulation(model, Δt=5minutes, stop_time=30days)
```

### CODE REVIEW

**Comment by glwagner** [comment]
I'm having trouble with a cooling case. (I hoped to use this to test sea ice fluxes). @simone-silvestri if you get a chance, can you take a look? I added the script here: https://github.com/CliMA/ClimaOcean.jl/blob/glw/idealized-atmosphere/examples/arctic_night.jl

It produces

https://github.com/user-attachments/assets/a5e70068-11ce-4342-82dc-130c835eb2c5

**Comment by simone-silvestri** [comment]
Weird that there is also a momentum stress but the velocity remains zero

**Comment by simone-silvestri** [comment]
Probably a hint, adding 
```julia

progress(simulation) = @info @sprintf("Time: %s, Δt: %s, surface (T, u): %.2e %.2e fluxes (T, u) %.2e %.2e\n", 
                                       prettytime(simulation.model.clock.time), prettytime(simulation.Δt),
                                       ocean.model.tracers.T[1, 1, grid.Nz], 
                                       ocean.model.velocities.u[1, 1, grid.Nz],
                                       ocean.model.tracers.T.boundary_conditions.top.condition[1, 1, 1], 
                                       ocean.model.velocities.u.boundary_conditions.top.condition[1, 1, 1])

simulation.callbacks[:progress] = Callback(progress, IterationInterval(100))
```
shows:
```julia
┌ Warning: This simulation will run forever as stop iteration = stop time = wall time limit = Inf.
└ @ Oceananigans.Simulations ~/.julia/packages/Oceananigans/dgGml/src/Simulations/simulation.jl:55
[ Info: Initializing simulation...
[ Info: Time: 0 seconds, Δt: 10 minutes, surface (T, u): -3.86e-02 0.00e+00 fluxes (T, u) 1.30e-04 -2.01e-04
[ Info:     ... simulation initialization complete (2.825 seconds)
[ Info: Executing initial time step...
┌ Warning: Simulation stopped during initialization.
└ @ Oceananigans.Simulations ~/.julia/packages/Oceananigans/dgGml/src/Simulations/run.jl:129
[ Info:     ... initial time step complete (1.101 seconds).
[ Info: Time: 16.667 hours, Δt: 10 minutes, surface (T, u): -1.06e-01 3.56e-02 fluxes (T, u) 0.00e+00 0.00e+00
[ Info: Time: 1.389 days, Δt: 10 minutes, surface (T, u): -1.06e-01 3.56e-02 fluxes (T, u) 0.00e+00 0.00e+00
[ Info: Time: 2.083 days, Δt: 10 minutes, surface (T, u): -1.06e-01 3.56e-02 fluxes (T, u) 0.00e+00 0.00e+00
[ Info: Time: 2.778 days, Δt: 10 minutes, surface (T, u): -1.06e-01 3.56e-02 fluxes (T, u) 0.00e+00 0.00e+00
[ Info: Time: 3.472 days, Δt: 10 minutes, surface (T, u): -1.06e-01 3.56e-02 fluxes (T, u) 0.00e+00 0.00e+00
[ Info: Time: 4.167 days, Δt: 10 minutes, surface (T, u): -1.06e-01 3.56e-02 fluxes (T, u) 0.00e+00 0.00e+00
[ Info: Time: 4.861 days, Δt: 10 minutes, surface (T, u): -1.06e-01 3.56e-02 fluxes (T, u) 0.00e+00 0.00e+00
[ Info: Time: 5.556 days, Δt: 10 minutes, surface (T, u): -1.06e-01 3.56e-02 fluxes (T, u) 0.00e+00 0.00e+00
[ Info: Time: 6.250 days, Δt: 10 minutes, surface (T, u): -1.06e-01 3.56e-02 fluxes (T, u) 0.00e+00 0.00e+00
[ Info: Time: 6.944 days, Δt: 10 minutes, surface (T, u): -1.06e-01 3.56e-02 fluxes (T, u) 0.00e+00 0.00e+00
[ Info: Time: 7.639 days, Δt: 10 minutes, surface (T, u): -1.06e-01 3.55e-02 fluxes (T, u) 0.00e+00 0.00e+00
[ Info: Time: 8.333 days, Δt: 10 minutes, surface (T, u): -1.06e-01 3.55e-02 fluxes (T, u) 0.00e+00 0.00e+00
[ Info: Time: 9.028 days, Δt: 10 minutes, surface (T, u): -1.06e-01 3.55e-02 fluxes (T, u) 0.00e+00 0.00e+00
[ Info: Time: 9.722 days, Δt: 10 minutes, surface (T, u): -1.06e-01 3.55e-02 fluxes (T, u) 0.00e+00 0.00e+00
```
Meaning the fluxes are somewhat inactive

**Comment by simone-silvestri** [comment]
Ok I think this is the problem:
https://github.com/CliMA/ClimaOcean.jl/blob/b11863fc5f1c6f8d7c78d22ed5118ce4bfea4fb1/src/OceanSeaIceModels/ocean_sea_ice_model.jl#L72

as soon as T < minimum temperature, the fluxes shut down.
With 
```julia
model  = OceanSeaIceModel(ocean, nothing; atmosphere, radiation)
```
we get

https://github.com/user-attachments/assets/dc748738-b9ce-4eb0-88fb-ad071a0f7967

**Comment by simone-silvestri** [comment]
there must be some bug though because clearly `0 > -1.8` that is the default "minimum_temperature". I ll take a look

**Comment by glwagner** [comment]
Nice find

**Comment by glwagner** [comment]
@navidcy this PR provides a path towards implementing a simple test for `FreezingLimitedOceanTemperature`. We can use the example here.

File: src/OceanSeaIceModels/PrescribedAtmospheres.jl [hunk 1]
```diff
@@ -314,36 +316,72 @@ function Base.show(io::IO, pa::PrescribedAtmosphere)
...
+                         metadata = nothing,
+                         reference_height = 10, # meters
+                         boundary_layer_height = 600 # meters,
+                         thermodynamics_parameters = PrescribedAtmosphereThermodynamicsParameters(FT),
                          velocities = nothing,
                          pressure = nothing,
                          freshwater_flux = nothing,
+                         auxiliary_freshwater_flux = nothing,
                          downwelling_radiation = nothing,
                          tracers = nothing)
```
Comment (simone-silvestri) [inline src/OceanSeaIceModels/PrescribedAtmospheres.jl:361]:
```suggestion
                              auxiliary_freshwater_flux = nothing,
                              velocities            = default_atmosphere_velocities(grid, times),
                              tracers               = default_atmosphere_tracers(grid, times),
                              pressure              = default_atmosphere_pressure(grid, times),
                              freshwater_flux       = default_freshwater_flux(grid, times),
                              downwelling_radiation = default_downwelling_radiation(grid, times))
```

**Review by simone-silvestri — APPROVED** [review]

**Pull request merged**